src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -384,31 +265,13 @@ public Credential makeCredential() {
     return cred;
   }
 
-  /**
-   * Performs the firing of listeners and the updates to the UI that are normally performed as part
-   * of a log in or log out, and retrieves any persistently stored credentials upon log in, but does
-   * not actually interact with an OAuth server. This method is provided for use in tests.
-   *
-   * @param login
-   *     {@code true} if a log in is to be simulated, {@code false} if a log out is to be simulated
-   */
-  public void simulateLoginStatusChange(boolean login) {
-    if (login) {
-      retrieveSavedCredentials();
-    }
-    notifyLoginStatusChange(login);
-    uiFacade.notifyStatusIndicator();
-  }
-
-  private void updateLoginState(GoogleTokenResponse tokenResponse) {
-    refreshToken = tokenResponse.getRefreshToken();
-    accessToken = tokenResponse.getAccessToken();
-    oAuth2Credential = makeCredential();
-    accessTokenExpiryTime = System.currentTimeMillis() / 1000 + tokenResponse.getExpiresInSeconds();
-    email = queryEmail();
-    persistCredentials();
-    uiFacade.notifyStatusIndicator();
-    notifyLoginStatusChange(true);
+  private void updateLoginState(GoogleTokenResponse tokenResponse)
+      throws IOException, EmailAddressNotReturnedException {
+    Credential credential = makeCredential(tokenResponse.getAccessToken(),
```

Comment:
Now you asked it, I have the same question. The original code didn't set it. Maybe the check is anyway being done on the server-side?